src/tensor_operations/matrix_decomposition.jl

```diff
@@ -573,7 +573,8 @@ function factorize_eigen(A::ITensor, Linds...; kwargs...)
     # (Lis..., prime(Lis)...)
     delta_A2 = replaceinds(delta_A2, Lis, dag(simLis))
     noprime!(delta_A2)
-    A2 += delta_A2
+    itensor(data(A2) .+= data(delta_A2), inds(A2))
```

Comment:
This will be wrong for sparse tensors, needs a fix in `NDTensors`. Like the ones above, I think a change to `permutedims` dispatching on `CuArray` like you did for `mul!` will be sufficient.

Reply:
Practically all of the code which uses `data` is just a test to see if these operations are correlated with the performance issues with CUDA. I can verify from these tests that the performance issues are caused by these operations and will shift to find a more practical implementation which solves the issues.

Reply:
Gotcha, makes sense. Glad it seems to be only a few operations that need to be optimized/dispatched properly, I'm sure we can find a simple lower level solution.